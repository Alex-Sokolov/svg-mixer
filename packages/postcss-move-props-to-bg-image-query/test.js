const postcss = require('postcss');

const plugin = require('.');

function exec(input, opts) {
  return postcss()
    .use(plugin(opts))
    .process(input, { from: __filename })
    .then(({ css }) => css);
}

it('should move svg-* props by default', async () => {
  expect(await exec(
    '.a {background: url(1.png); fill: red}'
  )).toMatchSnapshot();

  expect(await exec(
    '.a {background: url(1.png); svg-fill: red}'
  )).toMatchSnapshot();
});

it('should preserve if any query params already there', async () => {
  expect(await exec(
    '.a {background: url(1.png?qwe); svg-fill: red}'
  )).toMatchSnapshot();
});

it('should allow to override default props', async () => {
  expect(await exec(
    '.a {background: url(1.png); svg-fill: red; color: blue}',
    { props: ['color'] }
  )).toMatchSnapshot();
});

it('should allow to not remove props after moving them to query', async () => {
  expect(await exec(
    '.a {background: url(1.png); svg-fill: red;}',
    { remove: false }
  )).toMatchSnapshot();
});

it('should allow to use custom transformer', async () => {
  expect(await exec(
    '.a {background: url(1.png); svg-fill: red;}',
    { transform: ({ prop: name, value }) => ({ name: name.toUpperCase(), value }) }
  )).toMatchSnapshot();
});

it('should always encode prop value even with custom transformer', async () => {
  expect(await exec(
    '.a {background: url(1.png); svg-fill: #fff;}'
  )).toMatchSnapshot();

  expect(await exec(
    '.a {background: url(1.png); svg-fill: #fff;}',
    { transform: ({ prop: name, value }) => ({ name: name.toUpperCase(), value }) }
  )).toMatchSnapshot();
});

describe('match option', () => {
  it('should include properly', async () => {
    expect(await exec(
      '.a {background: url(1.svg); svg-fill: red;}',
      { match: ['*.png', '*.svg'] }
    )).toMatchSnapshot();
  });

  it('should skip properly', async () => {
    expect(await exec(
      '.a {background: url(300.svg); svg-fill: red;}',
      { match: ['*.svg', '!300.*'] }
    )).toMatchSnapshot();
  });

  it('should handle super complex cases', async () => {
    expect(await exec(`
.a {
  background: url(1.svg) url(2.svg) url(3.svg) url(foo.svg);
  background-image: url(1.svg) url(2.svg) url(3.svg) url(foo.svg);
  svg-fill: red;
}`,
    { match: /\d\.svg$/ }
    )).toMatchSnapshot();
  });
});
