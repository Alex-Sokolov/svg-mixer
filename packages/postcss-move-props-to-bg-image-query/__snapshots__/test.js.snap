// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match option should handle super complex cases 1`] = `
"
.a {
  background: url(1.svg?fill=red) url(2.svg?fill=red) url(3.svg?fill=red) url(foo.svg);
  background-image: url(1.svg?fill=red) url(2.svg?fill=red) url(3.svg?fill=red) url(foo.svg);
}"
`;

exports[`match option should include properly 1`] = `".a {background: url(1.svg?fill=red);}"`;

exports[`match option should skip properly 1`] = `".a {background: url(300.svg); svg-fill: red;}"`;

exports[`should allow to not remove props after moving them to query 1`] = `".a {background: url(1.png); svg-fill: red;}"`;

exports[`should allow to override default props 1`] = `".a {background: url(1.png); svg-fill: red; color: blue}"`;

exports[`should allow to use custom transformer 1`] = `".a {background: url(1.png); svg-fill: red;}"`;

exports[`should always encode prop value even with custom transformer 1`] = `".a {background: url(1.png); svg-fill: #fff;}"`;

exports[`should always encode prop value even with custom transformer 2`] = `".a {background: url(1.png); svg-fill: #fff;}"`;

exports[`should move svg-* props by default 1`] = `".a {background: url(1.png); fill: red}"`;

exports[`should move svg-* props by default 2`] = `".a {background: url(1.png); svg-fill: red}"`;

exports[`should preserve if any query params already there 1`] = `".a {background: url(1.png?qwe); svg-fill: red}"`;
